src/Delivery/Console/ClearCacheCommand.php

```diff
@@ -24,25 +24,23 @@ protected function configure()
             ->setName('delivery:cache:clear')
             ->setDefinition([
                 new InputArgument(
-                    'space-id', InputArgument::REQUIRED,
-                    'ID of the space to use.'
-                ),
-                new InputArgument(
-                    'cache-dir', InputArgument::REQUIRED,
-                    'The directory to write the cache to.'
+                    'cache-item-pool-factory-class', InputArgument::REQUIRED,
+                    'The FQCN of a class to be used as a cache item pool factory. Must implement \Contentful\Console\CacheItemPoolFactoryInterface.'
                 ),
             ]);
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $spaceId = $input->getArgument('space-id');
-        $cacheDir = $input->getArgument('cache-dir');
+        $cachePoolFactoryClass = $input->getArgument('cache-item-pool-factory-class');
+        $cacheItemPoolFactory = new $cachePoolFactoryClass();
+        if (!$cacheItemPoolFactory instanceof CacheItemPoolFactoryInterface) {
+            throw new \InvalidArgumentException("Cache item pool factory class must implement \Contentful\Console\CacheItemPoolFactoryInterface");
+        }
 
-        $cache = new FilesystemAdapter($spaceId, 0, $cacheDir);
-        $clearer = new CacheClearer($cache);
+        $clearer = new CacheClearer($cacheItemPoolFactory->getCacheItemPool());
```

Comment:
I would add a check here, to be sure that the result of `$cacheItemPoolFactory->getCacheItemPool()` actually is an instance of `CacheItemPoolInterface`.